teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -72,8 +91,331 @@ Per [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-
+
+However, the default value of `enable-api-fields` will **not** be changed and it will remain to be `beta`, so the **beta** features will continue to be enabled by default.
+
+- **Impacts on users:**
+    - Cluster operators:
+      - Current cluster operators with `enable-api-fields` set to `alpha` or `beta` should not experience any changes.
+      - This makes it possible for cluster operators to have full control over only stable feature usages, rather than user overrides. For example, currently if cluster operators want their users to only opt-in “stable” features, they cannot do so for `v1beta1` apiVersion for the exception of resolver, object params and results.
+    - Pipeline and Task authors: 
+    This affects pipeline users in the situation where cluster operators have chosen to set `enable-api-fields` to `stable` (i.e. those who have changed the current default `beta` value). 
+    This also affects users who are using beta features as default with the previous configuration of `enable-api-fields=stable`, like resolvers, in `v1beta1`. Such pipeline with beta features used in `v1beta1` apiVersion would not result in a successful pipelineRun because of [the validation after the conversion to `v1` as the storage version](https://github.com/tektoncd/pipeline/blob/d9d2d1760fa534e2dfb16ca656cdf2d293a5900e/pkg/apis/pipeline/`v1`/taskref_validation.go#L39). Since in this example, resolver is still a beta feature, this pipeline should not have been applied to the cluster that has not set `enable-api-fields=beta` in the first place.
+
+### Per feature flag for _new_ api-driven features
+Introduce per-feature flags for each **new** API driven feature. Each feature will have its own flag, instead of using the group API driven flag `enable-api-fields` to enable or disable all features of a stability level. Note that our proposal only takes effect on api-driven features while behavioural flags will remain the existing beahviour.
+
+The flag will also include the maintainer information and stability level for the feature as the source of truth. New behavioural features will also have a new per-feature flag to either enable or disable the feature. This will allow behavioural features that has values leading to behaviors in different stability levels to be turned on or off instead of depending on the stability level of features that are enabled.
+
+For example:
+
+```yaml
+apiVersion: `v1`
+kind: ConfigMap
+metadata:
+  name: feature-flags
+data:
+  # alpha: v0.53
+  #
+  # Pipeline in pipeline has ... functionalities.
+  # It is disabled by default.
+  enable-pipeline-in-pipeline: "false"
+  # alpha: v0.53
+  # 
+  # Trusted artifacts has ... functionalities.
+  # It is disabled by default.
+  enable-trusted-artifacts: "true"
```

Comment:
suggested additional comment: set it to true to enable this feature

Reply:
Thanks:) updated.